src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -1215,6 +1215,17 @@ class G1UpdateRegionsAfterRebuild : public HeapRegionClosure {
   }
 };
 
+class G1ObjectCountIsAliveClosure: public BoolObjectClosure {
+  G1CollectedHeap* _g1h;
+public:
+  G1ObjectCountIsAliveClosure(G1CollectedHeap* g1h) : _g1h(g1h) { }
+
+  bool do_object_b(oop obj) {
+    return obj != nullptr &&
+           (!_g1h->is_in_reserved(obj) || !_g1h->is_obj_dead(obj));
```

Comment:
I think the first clause can be removed (or asserted); the (parallel-)objectiterator for g1 always returns `true` for that as it only ever iterates over the heap regions.
Also the `!= nullptr` is imo removable for the same reason. We are walking the heap linearly, so I do not see how that value could be null.

Reply:
Think so; will do it in another PR.